jdk/make/CopyFiles.gmk

```diff
@@ -283,6 +283,14 @@ ifneq ($(OPENSSL_BUNDLE_LIB_PATH), )
     OPENSSL_TARGET_LIB = $(JDK_OUTPUTDIR)/lib$(OPENJDK_TARGET_CPU_LIBDIR)/$(OPENSSL_LIB_NAME)
   endif
 
+  # On Mac OS, update the crypto library path to @rpath as the default is /usr/local/lib/
+  ifeq ($(OPENJDK_BUILD_OS), macosx)
+    .PHONY : RPATH_CRYPTO_LIB
+    TARGETS += RPATH_CRYPTO_LIB
```

Comment:
I think this needs to be `COPY_FILES` instead of `TARGETS` so there's a dependency on `RPATH_CRYPTO_LIB`, otherwise the command below will not execute. See `jdk/make/CompileNativeLibraries.gmk`:
> all: $(COPY_FILES) $(BUILD_LIBRARIES)

Reply:
Confirmed in https://ci.eclipse.org/openj9/job/Build-JDK8-osx_x86-64_cmprssptrs/88/:
```
$ objdump -dylib-id -macho jre/lib/libcrypto.1.1.dylib
jre/lib/libcrypto.1.1.dylib:
/usr/local/lib/libcrypto.1.1.dylib
```

Reply:
Even if I run `install_name_tool` manually, it doesn't fix the problem because `libjncrypto.dylib` was created before the right id is installed in `libcrypto.dylib`. I think the right time to run `install_name_tool` is during configuration - after the file is copied to `${LOCAL_CRYPTO}`.

Reply:
My previous comment applies when configured using `--with-openssl={path}`; when configure decides `BUILD_OPENSSL=yes`, then `install_name_tool` should be invoked within `closed/openssl.gmk`.

Reply:
I had been assuming the changes to `jdk/make/closed/autoconf/custom-hook.m4` were effective, but they won't be until you run `bash common/autoconf/autogen.sh` and push the new generated files to your branch.

Reply:
Thank you @keithc-ca for your comments. As you identified, I missed to update generated-configure.sh (because of the influence of working with openjdk11 as it's automatically generated while building :-)). Sorry for the inconvenience.
@pshipton I have updated generated-configure.sh, squashed and pushed. Please resubmit the build jobs.

Reply:
I'm not convinced that, as merged, we are guaranteed that `install_name_tool` will be used to update `libcrypto.dylib` *before* `libjncrypto.dylib` is created and so we can't be sure it will have the proper reference to `libcrypto.dylib`. Unfortunately, I also don't have a Mac that is capable of building jdk8.